main.py

```diff
@@ -1,67 +1,103 @@
-                    and jsonData["flow_category"] != "dry_flow"
-                ):
-                    handle_flow_category_data(jsonData)
-
-                    # Handle call logs
-                    calllog_service = services.CallLogService()
-                    calllog_service.handle_call_log(jsonData)
-
-                # All the prompt responses are captured with results
-                if "results" in jsonData:
-                    handle_prompts(jsonData)
-
-                # Handle content details
-                user_module_content_id = None
-                if "content_id" in jsonData:
-                    content_service = services.ContentService()
-                    user_module_content_id = content_service.add_user_module_content(
-                        jsonData
-                    )
-                if user_module_content_id:
-                    calllog_service.update_user_module_content_id_in_call_log(
-                        user_module_content_id
-                    )
-
-                # Handle contact groups
-                if (
-                    "groups" in jsonData["contact"]
-                    and jsonData["contact"]["groups"] is not None
-                    and jsonData.get("flow_category", None) == "dry_flow"
-                ):
-                    handle_user_group_data(jsonData)
-
-                # Handle custom fields
-                if (
-                    "fields" in jsonData["contact"]
-                    and jsonData["contact"]["fields"] is not None
-                    and jsonData.get("flow_category", None) == "dry_flow"
-                ):
-                    handle_user_custom_field_data(jsonData)
+                webhook_log = transaction_log_service.create_new_webhook_log(jsonData)
+            processed = handle_payload(jsonData)
 
-                transaction_log_service.mark_webhook_log_as_processed(webhook_log)
-            else:
+            if processed is False:
+                return jsonify(message="Something went wrong!"), 400
+            elif processed == -1:
                 return jsonify(message="Contact"), 400
-        except:
-            return jsonify(message="Something went wrong!"), 400
+
+            if "contact" in jsonData:
+                transaction_log_service.mark_webhook_log_as_processed(webhook_log)
+        elif jsonData and jsonData.get("type", None) == "retry_failed_log":
+            retry_failed_webhook(transaction_log_service)
+
         return jsonify(message="Success"), 200
     else:
         return jsonify(message="Currently, the system do not accept a GET request"), 405
 
 
+def retry_failed_webhook(transaction_log_service):
+    failed_webhook_logs = transaction_log_service.get_failed_webhook_transaction_log()
+
+    for log in failed_webhook_logs:
+        json_data = json.loads(log.payload)
+        json_data["log_created_on"] = log.created_on
+        processed = handle_payload(json_data, True)
```

Comment:
Processing all the failed logs. The second parameter is for differentiating regular webhooks and retry webhooks